code/modules/mob/living/simple_animal/slaughter/slaughter.dm

```diff
@@ -64,26 +72,44 @@
 /mob/living/simple_animal/slaughter/phasein()
 	. = ..()
 	speed = 0
-	boost = world.time + 30
+	boost = world.time + 60
 
 
-//////////The Loot
-
-/obj/item/weapon/demonheart
-	name = "demon's heart"
-	desc = "It's still faintly beating with rage"
+//The loot from killing a slaughter demon - can be consumed to allow the user to blood crawl
+/obj/item/organ/internal/heart/demon
+	name = "demon heart"
+	desc = "Still it beats furiously, emanating an aura of utter hate."
 	icon = 'icons/obj/surgery.dmi'
-	icon_state = "heart-on"
+	icon_state = "demon_heart"
 	origin_tech = "combat=5;biotech=8"
 
-/obj/item/weapon/demonheart/attack_self(mob/living/user)
-	visible_message("[user] feasts upon the [src].")
+/obj/item/organ/internal/heart/demon/attack(mob/M, mob/living/carbon/user, obj/target)
+	if(M != user)
+		return ..()
+	user.visible_message("<span class='warning'>[user] raises [src] to their mouth and tears into it with their teeth!</span>", \
+						 "<span class='danger'>An unnatural hunger consumes you. You raise [src] your mouth and devour it!</span>")
+	playsound(user, 'sound/magic/Demon_consume.ogg', 50, 1)
 	for(var/obj/effect/proc_holder/spell/knownspell in user.mind.spell_list)
 		if(knownspell.type == /obj/effect/proc_holder/spell/bloodcrawl)
-			user <<"<span class='notice'>You already know how to blood crawl.</span>"
+			user <<"<span class='warning'>...and you don't feel any different.</span>"
 			qdel(src)
 			return
-	user << "You absorb some of the demon's power!"
-	user.mind.AddSpell(new /obj/effect/proc_holder/spell/bloodcrawl)
-	qdel(src)
+	user.visible_message("<span class='warning'>[user]'s eyes flare a deep crimson!</span>", \
+						 "<span class='userdanger'>You feel a strange power seep into your body... you have absorbed the demon's blood-travelling powers!</span>")
+	for(var/obj/item/organ/internal/O in user.internal_organs)
+		if(istype(O, /obj/item/organ/internal/heart))
+			O.Remove(user, 1)
+			O.loc = get_turf(user)
+			qdel(O)
```

Comment:
So if I just cut this whole if(istype) section out and replace it with the Insert() it shouldn't break the game?